sktime/utils/_testing/deep_equals.py

```diff
@@ -93,7 +93,12 @@ def ret(is_equal, msg):
                 msg = ""
             return ret(index_equal and values_equal, msg)
         else:
-            return ret(x.equals(y), f".series_equals, x = {x} != y = {y}")
+            # this construction is needed so costly str(pd.Series) is only
```

Comment:
Alternatively, we could try to make `ret` lazy evaluating capable. The idea would be something as: 

```python
  def ret(is_equal, msg="", string_arguments=None):
      string_arguments = [] if len(string_arguments) == 0 else string_arguments
      if return_msg:
          if is_equal:
              msg = ""
          elif len(string_arguments) > 0:
              msg = msg.format(*string_arguments)
          return is_equal, msg
      else:
          return is_equal
```

this would enable to call `ret(x.equals(y), ".index_equals, x = {} != y = {}", [x, y])`. Consequently, we also would not require this construction.

Reply:
hm, I see! Great idea!

My main worry is that it doesn't actually work precisely as described, there rare a lot of case distinctions, e.g., `string_arguments = None`, and calling `len(None)`, etc.

I don't mind if you try to make this change, we need to check that the negative case is also tested somewhere.

Reply:
I can try to change it. Should I create a new PR for this? Or work on this?

Reply:
either is fine with me, as you prefer!

Reply:
I added a new PR based on this branch so that we could merge this PR, if we prefer this solution.

New PR is here: #5044